ansible/cloud_providers/osp_infrastructure_deployment.yml

```diff
@@ -0,0 +1,167 @@
+---
+- name: Step 001.1 Deploy Infrastructure
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  tags:
+    - step001
+    - step001.1
+    - deploy_infrastructure
+  tasks:
+    - name: Run infra-osp-template-generate Role
+      import_role:
+        name: infra-osp-template-generate
+
+    - name: Run infra-osp-template-create Role
+      import_role:
+        name: infra-osp-template-create
+
+- name: Step 001.2 Create Inventory and SSH config setup
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: false
+  tags:
+    - step001
+    - step001.2
+    - create_inventory
+    - create_ssh_config
+  environment:
+    OS_AUTH_URL: "{{ osp_auth_url }}"
+    OS_USERNAME: "{{ osp_auth_username }}"
+    OS_PASSWORD: "{{ osp_auth_password }}"
+    OS_PROJECT_NAME: "{{ osp_project_name }}"
+    OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
+    OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+  tasks:
+    - name: Gather instance facts
+      os_server_facts:
+        server: "*"
+        # filters:
+        #   project: "{{ project_tag }}"
+      register: r_osp_facts
+
+    - name: debug osp_facts
+      debug:
+        var: r_osp_facts
+        verbosity: 2
+
+    - debug:
+        msg: The floating IP for {{ item }} is {{ r_osp_facts | json_query(query) }}
+      vars:
+        query: "ansible_facts.openstack_servers[?@.name=='{{ item }}'].public_v4|[0]"
+      with_items:
+        - "bastion"
+
+    - name: Create DNS entries
+      debug:
+        msg: Currently using {{ ocp_cluster_dns_zone }} on server {{ ocp_cluster_dns_server }}
+
+    - name: Add DNS entry for bastion
+      nsupdate:
+        server: "{{ ocp_cluster_dns_server }}"
+        zone: "{{ ocp_cluster_dns_zone }}"
+        record: bastion.{{ guid }} #.{{ ocp_cluster_dns_zone }}
+        type: A
+        ttl: 5
+        value: "{{ r_osp_facts | json_query(query) }}"
+        key_name: "{{ ddns_key_name }}"
+        key_secret: "{{ ddns_key_secret }}"
+      vars:
+        query: "ansible_facts.openstack_servers[?@.name=='bastion'].public_v4|[0]"
+
+    # - name: Get HOSTEDZONEID
```

Comment:
I think we can remove all the AWS stuff.

Reply:
agreed